howso/update_cases.amlg

```diff
@@ -440,90 +440,149 @@
 					mode "robust"
 					weight_feature ".none"
 				))
-			;default value of 1 for the accumulate_weight_feature
-			new_weight_label_and_value (zip_labels (list accumulate_weight_feature) (list 1))
 			weight_feature_index
 				;will retrieve the index of weight feature or (null)
 				(if (contains_value features accumulate_weight_feature)
 					(get (zip features (indices features)) accumulate_weight_feature)
 				)
 			context_features (filter (lambda (!= (current_value) accumulate_weight_feature)) features)
+			num_cases (size cases)
+			batch_size 0
+			case_index 0
+			batch_of_cases []
+			closest_cases_maps {}
+			entropy_weights []
+		))
+		(declare (assoc k_param (get hyperparam_map "k") ))
+		(declare (assoc
+			;cache the maximum entropy possibe given k similar cases, used to compute the entropy ratio for each accumulated case
+			max_influence_entropy (entropy (range (/ 1 k_param) 1 k_param 1))
+			;smallest possible amount to be distributed
+			smallest_entropy_amount (/ 1 (call !GetNumTrainingCases))
 		))
 
-		(map
-			(lambda (let
-				(assoc
-					case_value_map (zip features (current_value 1))
-					closest_cases_map (assoc)
-					total_weight 0
-					case_weight
-						(if (!= (null) weight_feature_index)
-							(get (current_value 1) weight_feature_index)
-							;assume weight of 1 if weight feature values are not given
-							1
-						)
-				)
+		;accumulate all cases in batches of !ablatedCasesDistributionBatchSize
+		(while (< case_index num_cases)
+			(assign (assoc batch_size (min !ablatedCasesDistributionBatchSize (- num_cases case_index)) ))
+			(assign (assoc
+				batch_of_cases (unzip cases (range case_index (+ case_index batch_size -1)))
+			))
 
-				(assign (assoc
-					;map of case_id -> weight
-					closest_cases_map
-						(compute_on_contained_entities (list
-							(query_nearest_generalized_distance
-								(get hyperparam_map "k")
-								context_features
-								(unzip case_value_map context_features)
-								(get hyperparam_map "featureWeights")
-								!queryDistanceTypeMap
-								(get hyperparam_map "featureDomainAttributes")
-								(get hyperparam_map "featureDeviations")
-								(get hyperparam_map "p")
-								(get hyperparam_map "dt")
-								(if (!= weight_feature ".none") weight_feature (null))
-								;use a fixed random seed to guarantee deterministic behavior for reacts (named "fixed rand seed")
-								"fixed rand seed"
-								(null) ;radius
-								!numericalPrecision
+			;store all the local models for the cases in one pass
+			(assign (assoc
+				closest_cases_maps
+					||(map
+						(lambda (let
+							(assoc case_value_map (zip features (current_value 1)) )
```

Comment:
It seems to me like `case_value_map` is only used on line 485 and is immediately unzipped. Couldn't we just drop the `(let` and just use `(current_value` directly in the query?